tests/playwright-test/config.spec.ts

```diff
@@ -259,33 +260,31 @@ test('should filter by project and parse as RegExp', async ({ runInlineTest }) =
       test('one', async ({}) => {
         console.log('%%' + test.info().project.name);
       });    `
-  }, { project: '.*oj.*t-Na.?e' });
+  }, { '--project-grep': '.*oj.*t-Na.?e' });
   expect(result.exitCode).toBe(0);
   expect(result.output).toContain('Running 1 test using 1 worker');
   expect(new Set(result.outputLines)).toEqual(new Set([
     'project-name',
   ]));
 });
 
-test('should filter by project and only match if its full-match', async ({ runInlineTest }) => {
-  const result = await runInlineTest({
-    'playwright.config.js': `
-      module.exports = {
-        projects: [
-         { name: 'prefix-foobar-suffix' },
-         { name: 'foobar' }
-        ]
-      };
+test('should print nice error when some of the project greps do not match anything', async ({ runInlineTest }) => {
+  const { output, exitCode } = await runInlineTest({
+    'playwright.config.ts': `
+      module.exports = { projects: [
+        { name: 'suite1' },
+        { name: 'suite2' },
+      ] };
     `,
-    'a.test.js': `
-      const { test } = require('@playwright/test');
-      test('one', async ({}) => {
-        console.log('%%' + test.info().project.name);
-      });    `
-  }, { project: 'foobar' });
-  expect(result.exitCode).toBe(0);
-  expect(result.output).toContain('Running 1 test using 1 worker');
-  expect(new Set(result.outputLines)).toEqual(new Set(['foobar']));
+    'a.test.ts': `
+      import { test, expect } from '@playwright/test';
+      test('pass', async ({}, testInfo) => {
+        console.log(testInfo.project.name);
+      });
+    `
+  }, { '--project-grep': ['aaa', 'bbb', 'SUite1'] });
+  expect(exitCode).toBe(1);
+  expect(output).toContain('Project(s) matching "aaa", "bbb" not found. Available named projects: "suite1", "suite2"');
```

Comment:
Why is this an error? Perhaps only show an error when no projects matched at all?

Reply:
The idea is to do the same as with --project, i.e. each specified regex should match something. But with single regex it should not be a problem.